src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_x509.c

```diff
@@ -287,3 +294,58 @@ static int32_t PopulateByteArray(JNIEnv* env, jbyteArray source, uint8_t* dest,
     (*env)->GetByteArrayRegion(env, source, 0, bytesLen, (jbyte*)dest);
     return CheckJNIExceptions(env) ? FAIL : SUCCESS;
 }
+
+static size_t min_size_t(size_t x, size_t y)
+{
+    return (x < y) ? x : y;
+}
+
+static void FindCertStart(const uint8_t** buffer, int32_t* len)
+{
+    assert(buffer != NULL && *buffer != NULL);
+    assert(*len >= 0);
+
+    if (iscntrl(**buffer) && !isspace(**buffer))
+    {
+        // If the character is a control character that isn't whitespace, then we're probably using a DER encoding
+        // and not using a PEM encoding in ASCII.
+        return;
+    }
+
+    const uint8_t* bufferLocal = *buffer;
+    int32_t lengthLocal = *len;
+
+    while (lengthLocal > 0)
+    {
+        const char pemHeader[] = "-----BEGIN ";
+        // Skip until we see the - that could start a PEM block.
+        while(lengthLocal > 0 && (!iscntrl(*bufferLocal) || isspace(*bufferLocal)) && *bufferLocal != pemHeader[0])
+        {
+            bufferLocal += 1;
+            lengthLocal -= 1;
+        }
+        
+        if (lengthLocal == 0 || (iscntrl(*bufferLocal) && !isspace(*bufferLocal)))
+        {
+            // Either the buffer is empty or we encountered a control character that isn't whitespace.
+            // In the empty case, we didn't find the PEM header, so we can't skip to it.
+            // In the control character case, we know that this isn't explanatory info since that needs to
+            // all be printable or whitespace characters, not non-whitespace control characters.
+            return;
+        }
+        
+        if (memcmp(bufferLocal, pemHeader, min_size_t((size_t)lengthLocal, sizeof(pemHeader) - 1)) == 0)
```

Comment:
Hm.  Could you avoid the need for min_size_t if you just bailed earlier once `lengthLocal <= sizeof(pemHeader)` (vs `== 0`)?

The absolute minimum is something I don't really want to compute, but it's at least `-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----`, which I count as 53 characters.  (The minimum value for a Base64 payload that could plausibly be a certificate is the part I don't want to do right now :smile:)

Reply:
I can add the earlier bailout.

Do we only need to handle the "BEGIN CERTIFICATE" case, or do we need to also be able to handle the other headers for PEM envelopes (like `X509 CRL`)?

Reply:
Well, `X509 CRL` wouldn't be an appropriate match here, but `CMS` might, so I guess the minimum is only somewhere above 20 (hypens) + 2 (spaces) + 6 (2x "CMS".Length) + 5 (BEGIN) + 3 (END) => 36.

But using sizeof(pemHeader) is an obvious "the next part can't match", so it's good enough.